### PR TITLE
Caas

### PR DESCRIPTION
**Pull request by mel-splunk**
very different from datadog and also reworking directories organization.

the goal is to have a working v1 for every "mandatory" alerts on managed kubernetes.

more "optional" alerts will be added in future PR, same for crucial alerts for non managed kuberntes only (like apiserver ...)

### CODE REVIEW

File: caas/docker/variables.tf [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  type        = string
+  default     = "20m"
+}
+
+# Memory_used detectors
+
+variable "memory_used_disabled" {
+  description = "Disable all alerting rules for memory_used detector"
+  type        = bool
+  default     = null
```
Comment (xp-1000) [inline caas/docker/variables.tf:64]: should be disabled

https://github.com/claranet/terraform-datadog-monitors/blob/master/caas/docker/inputs.tf#L86

File: caas/kubernetes/workload/detectors-k8s-workload.tf [hunk 2]
```diff
@@ -0,0 +1,82 @@
+resource "signalfx_detector" "heartbeat" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Kubernetes API server heartbeat"
```
Comment (pdecat) [inline caas/kubernetes/workload/detectors-k8s-workload.tf:2]: Same name as https://github.com/claranet/terraform-signalfx-detectors/pull/23/files#diff-7d36032359e68de3850038e41edeae9cR2
```suggestion
	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Kubernetes workload heartbeat"
```

File: caas/docker/variables.tf (hunk 1 above)
Comment (ybjosephson) [reply caas/docker/variables.tf:64]: updated

File: caas/kubernetes/workload/detectors-k8s-workload.tf (hunk 2 above)
Comment (ybjosephson) [reply caas/kubernetes/workload/detectors-k8s-workload.tf:2]: updated

File: caas/kubernetes/workload/detectors-k8s-workload.tf [hunk 3]
```diff
@@ -0,0 +1,82 @@
+resource "signalfx_detector" "heartbeat" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Kubernetes workload heartbeat"
+
+	program_text = <<-EOF
+		from signalfx.detectors.not_reporting import not_reporting
+		signal = data('kubernetes.job.failed', ${module.filter-tags.filter_custom})
```
Comment (pdecat) [inline caas/kubernetes/workload/detectors-k8s-workload.tf:6]:
```suggestion
		signal = data('kubernetes.job.active', ${module.filter-tags.filter_custom})
```

**Review by pdecat — DISMISSED** [review]
Here's a batch of fixes that I'm currently testing on GKE with the following agent configuration:

```
      globalDimensions:
        kubernetes_cluster: ${var.cluster_name}
        env: ${var.general["environment"]}
        sfx_monitored: true
      observers:
      - type: k8s-api

      monitors:

      # Collects cadvisor metrics through a Kubernetes kubelet instance via the /stats/container endpoint
      - type: kubelet-stats
        kubeletAPI:
          authType: serviceAccount
        datapointsToExclude:
        - dimensions:
            container_image:
            - '*pause-amd64*'
            - 'k8s.gcr.io/pause*'
          metricNames:
            - '*'
            - '!*network*'

      # Collects container metrics from a Kubernetes kubelet instance via the /stats/summary endpoint
      # Supposed to replace kubelet-stats but not yet complete
      - type: kubelet-metrics

      # Collects kube-apiserver metrics in Prometheus format
      - type: kubernetes-apiserver
        host: metrics-server.kube-system
        port: 443
        useHTTPS: true
        skipVerify: true
        useServiceAccount: true
        extraMetrics:
          - apiserver_request_total

      # Collects cluster-level metrics from the Kubernetes API server
      - type: kubernetes-cluster
        extraMetrics:
          - kubernetes.daemon_set.updated
          - kubernetes.deployment.updated
          - kubernetes.job.active
          - kubernetes.job.failed
          - kubernetes.stateful_set.updated

      # Collects usage stats about volumes mounted to Kubernetes pods (e.g. free space/inodes)
      - type: kubernetes-volumes
```

File: caas/kubernetes/pod/detectors-k8s-pod.tf [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	rule {
+		description           = "is too high > ${var.pod_phase_status_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.pod_phase_status_disabled_warning, var.pod_phase_status_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.pod_phase_status_notifications_warning, var.pod_phase_status_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
```
Comment (pdecat) [inline caas/kubernetes/pod/detectors-k8s-pod.tf:48]:
```suggestion
}
```

**Comment by pdecat** [comment]
Rebased on master to check terraform fmt with CI.

**Review by xp-1000 — APPROVED** [review]

**Review by jphilaine — APPROVED** [review]

**Pull request merged**